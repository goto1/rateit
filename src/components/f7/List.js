import React from "react";
import PropTypes from "prop-types";

const List = ({ inset = false, className, children }) => {
  const classNames = `list-block ${inset ? "inset" : ""} ${className
    ? className
    : ""}`.trim();
  return (
    <div className={classNames}>
      <ul>
        {children}
      </ul>
    </div>
  );
};

List.propTypes = {
  inset: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default List;
