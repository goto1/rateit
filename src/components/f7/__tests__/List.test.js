import React from "react";
import List from "../List";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("List component", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <List>
          <span>HelloWorld!</span>
        </List>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render a span element", () => {
    const wrapper = shallow(
      <List>
        <span>HelloWorld!</span>
      </List>
    );
    const actual = wrapper.find("span").length;
    const expected = 1;
    expect(actual).toBe(expected);
  });

  it("should render with a custom styling class", () => {
    const wrapper = shallow(<List className="custom">HelloWorld!</List>);
    const actual = wrapper.props().className.includes("custom");
    const expected = true;
    expect(actual).toBe(expected);
  });

  it("should render with an inset styling class", () => {
    const wrapper = shallow(<List inset={true}>HelloWorld!</List>);
    const actual = wrapper.props().className.includes("inset");
    const expected = true;
    expect(actual).toBe(expected);
  });
});
