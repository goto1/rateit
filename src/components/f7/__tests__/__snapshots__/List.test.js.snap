// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render correctly 1`] = `


<div
  className="list-block"
>
  <ul>
    <span>
      HelloWorld!
    </span>
  </ul>
</div>
`;
